### DISCUSSION TITLE
What's the status and plans for backward compatibility?

### DISCUSSION THREAD

**Discussion by dpc** [n0-computer/iroh#2428] [Q&A] [answered]
We're thinking about using Iroh in Fedimint one day to get rid of DNS and also requirement for servers to have publically available IP.

While we can accept many shortcomings we are wondering: if we were to use it now, and implementation being used in many client mobile applications etc. will such clients be able to connect to the future versions of other Iroh clients.

The other requirement is Wasm support, but I'm aware there are existing issues about it, and we probably could work it around via other ways.

**Comment by matheus23** [comment, marked as answer]
> will such clients be able to connect to the future versions of other Iroh clients.

We're not yet in backwards comatibility mode. We'll have another (protocol-) breaking release.
That said,
- the upgrade will likely only involve minor API changes if at all
- some of the features we're building today are already built with backwards compatibility in mind (e.g. the current websocket work), so until 1.0 hits, we're *probably* not breaking things either
- we have plans on getting protocol-stable around October.

**Reply by dpc** [reply]
The API changes are not a problem for us at all, but we need to maintain compatibility of the protocol with the older devices, at least with certain rolling window of let's say at least 6 months.

I'm happy to hear about October estimate.

**Comment by matheus23** [comment]
> The other requirement is Wasm support, but I'm aware there are existing issues about it, and we probably could work it around via other ways.

I'm currently working on Wasm support, so I'd like to know more about your use case: Wasm support as in running in the browser, in nodejs/deno or in wasmtime?
What I'm working on at the moment is browser support.

**Reply by dpc** [reply]
We're only concerned about running in the browser. Ideally both from WASM-based PWA and just JS from the browser (possibly via WASM-based library).

**Reply by matheus23** [reply]
This is a blog post from yesterday, by the way: https://iroh.computer/blog/iroh-and-the-web
